src/SSHDebugPS/ConnectionManager.cs

```diff
@@ -1,31 +1,68 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-using Microsoft.SSHDebugPS.VS;
-using Microsoft.VisualStudio.Shell;
-using Microsoft.VisualStudio.Shell.Interop;
 using System;
 using System.Globalization;
 using System.Linq;
 using liblinux;
 using liblinux.Persistence;
-using Microsoft.DebugEngineHost;
-using Task = System.Threading.Tasks.Task;
+using Microsoft.SSHDebugPS.Docker;
+using Microsoft.SSHDebugPS.SSH;
+using Microsoft.SSHDebugPS.VS;
 using Microsoft.VisualStudio.Linux.ConnectionManager;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
 
 namespace Microsoft.SSHDebugPS
 {
     internal class ConnectionManager
     {
-        internal static Connection GetInstance(string name, ConnectionReason reason)
+        public static Connection GetDockerConnection(string name)
+        {
+            Connection remoteConnection = null;
+            DockerTransportSettings settings = null;
+            // Assume format is <server>/<container> where if <server> is specified, it is for SSH
+            string[] connectionStrings = name.Split('/');
+
+            string displayName;
+            string containerName;
+            if (connectionStrings.Length == 1)
+            {
+                // local connection
+                containerName = connectionStrings[0];
+                settings = new DockerExecShellSettings(containerName, isUnix: false);
+                displayName = name;
+            }
+            else if (connectionStrings.Length == 2)
+            {
+                string remoteConnectionString = connectionStrings[0];
+                containerName = connectionStrings[1];
+                remoteConnection = GetSSHConnection(remoteConnectionString);
+
+                // If SSH connection dialog was cancelled, we should cancel this connection.
+                if (remoteConnection == null)
+                    return null;
+                
+                settings = new DockerExecShellSettings(containerName, isUnix: true); // assume all remote is Unix for now.
+                displayName = remoteConnection.Name + '/' + containerName;
+            }
+            else
+            {
+                throw new ArgumentException("Argument format is incorrect");
```

Comment:
>ArgumentException("Argument format is incorrect"); [](start = 26, length = 50)

Does this get shown to the user?

Reply:
Might consider including the correct format in the error message so users know what to do, e.g. "Expected 'container-id' or 'remote-host/container-id'"

---
In reply to: [253703674](https://github.com/Microsoft/MIEngine/pull/831#discussion_r253703674) [](ancestors = 253703674)

Reply:
No it doesn't. I'm returning null (no connection) for now and added a comment that we will use the dialog in this instance.